pennylane/ops/op_math/controlled.py

```diff
@@ -864,9 +864,13 @@ def _decompose_custom_ops(op: Controlled) -> list["operation.Operator"]:
         # has some special case handling of its own for further decomposition
         return _decompose_pauli_x_based_no_control_values(op)
 
-    if isinstance(op.base, qml.GlobalPhase) and len(op.control_wires) == 1:
+    if isinstance(op.base, qml.GlobalPhase):
         # use Lemma 5.2 from https://arxiv.org/pdf/quant-ph/9503016
-        return [qml.PhaseShift(phi=-op.data[0], wires=op.control_wires)]
+        phase_shift = qml.PhaseShift(phi=-op.data[0], wires=op.control_wires[-1])
+        if len(op.control_wires) > 1:
+            phase_shift = ctrl(phase_shift, control=op.control_wires[:-1])
+        return phase_shift
```

Comment:
```suggestion
        phase_shift = qml.PhaseShift(phi=-op.data[0], wires=op.control_wires[-1])
        if len(op.control_wires) > 1:
            phase_shift = ctrl(phase_shift, control=op.control_wires[:-1])
        return [phase_shift]
```

Reply:
Ah shoot, thanks for adding this! I had it right and then pylint was complaining about too many return statements... :D